sdks/java/core/src/main/java/org/apache/beam/sdk/PipelineResult.java

```diff
@@ -34,6 +38,26 @@
   State getState();
 
   /**
+   * Cancels the pipeline.
+   * @throws IOException if there is a problem executing the cancel request.
```

Comment:
`@throws UnsupportedOperationException if the runner does not support cancelation.`